assets/js/modules/analytics/setup/setup-main.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Analytics Main setup component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import SetupForm from './setup-form';
+import ProgressBar from '../../../components/progress-bar';
+import { SvgIcon } from '../../../util';
+import { STORE_NAME, ACCOUNT_CREATE } from '../datastore/constants';
+import {
+	AccountCreate,
+	ErrorNotice,
+	ExistingTagError,
+} from '../common';
+const { useSelect } = Data;
+
+export default function SetupMain( { finishSetup } ) {
+	const accounts = useSelect( ( select ) => select( STORE_NAME ).getAccounts() );
+	const accountID = useSelect( ( select ) => select( STORE_NAME ).getAccountID() );
+	const hasExistingTag = useSelect( ( select ) => select( STORE_NAME ).hasExistingTag() );
+	const existingTagPermission = useSelect( ( select ) => select( STORE_NAME ).hasExistingTagPermission() );
+	const isFetchingAccounts = useSelect( ( select ) => select( STORE_NAME ).isFetchingAccounts() );
+	const isDoingSubmitChanges = useSelect( ( select ) => select( STORE_NAME ).isDoingSubmitChanges() );
+	const hasResolvedAccounts = useSelect( ( select ) => select( STORE_NAME ).hasFinishedResolution( 'getAccounts' ) );
+	const isCreateAccount = ACCOUNT_CREATE === accountID;
+
+	let viewComponent;
+	if ( isFetchingAccounts || isDoingSubmitChanges || ! hasResolvedAccounts ) {
```

Comment:
This may be worth an inline comment for (I'll add one) but the reason this checks for `hasResolvedAccounts` as well is because without it there is a substantial flicker when the component initially loads before it starts loading accounts where it would otherwise show the `AccountCreate` case because there were no accounts yet.